### PR TITLE
update documentation

### PR DESCRIPTION
**Pull request by note**
* references to "ssl-config.ssl.\*" config properties updated to "ssl-config.\*" in .rst files
* minor typos in docs fixed

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -24,6 +24,7 @@ lazy val sslConfigCore = project.in(file("ssl-config-core"))
         case "2.10" => Seq.empty[ModuleID]
         case _      => Seq(Library.parserCombinators)
       }),
+    libraryDependencies ++= Dependencies.testDependencies,
```
Comment (note) [inline build.sbt]: That's why build was failing

File: documentation/src/sphinx/WSQuickStart.rst [hunk 2]
```diff
@@ -126,5 +126,4 @@ works with:
 
     WS.url("https://example.com").get()
 
-You can see more examples on the [[example
-configurations\|ExampleSSLConfig]] page.
+You can see more examples on the :ref:`example configurations\ <ExampleSSLConfig>` page.
```
Comment (ktoso) [inline documentation/src/sphinx/WSQuickStart.rst]: are we sure this is correct?

File: documentation/src/sphinx/Protocols.rst [hunk 3]
```diff
@@ -30,15 +30,15 @@ you can set it specifically in the client:
...
+    ssl-config.protocol = "TLSv1.2"
 
 If you want to define the list of enabled protocols, you can do so
 explicitly:
 
 ::
 
     # passed into sslContext.getDefaultParameters().setEnabledProtocols()
-    ssl-config.ssl.enabledProtocols = [
+    ssl-config.enabledProtocols = [
```
Comment (ktoso) [inline documentation/src/sphinx/Protocols.rst]: thanks, great catch everywhere

**Comment by ktoso** [comment]
LGTM, thanks!

File: documentation/src/sphinx/WSQuickStart.rst (hunk 2 above)
Comment (note) [inline documentation/src/sphinx/WSQuickStart.rst]: You mean the syntax itself? Yes, it works (I tested it locally) and it's used in many other places in this project

Comment (ktoso) [inline documentation/src/sphinx/WSQuickStart.rst]: The previously here `|` was a bit weird, so got curious.
Thanks for checking :)

**Pull request merged**